### PR TITLE
feat: enable linting for floating promises.

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR adds ESLinting for floating promises. Floating promises are defined as any function that resolves an unhandled promise. A handled promise can either be returned directly, awaited, then/catch'd, or voided.

This change would have immediately solved https://github.com/across-protocol/relayer-v2/commit/723bdcfae365051fcb05a089ebf4e6fc998bb2b0. We can now enforce that un-awaited promises will never be uncaught again.

### CODE REVIEW

File: test/UBAClient.validateFlow.ts [hunk 1]
```diff
@@ -422,11 +422,11 @@ describe("UBAClient: Flow validation", function () {
           // Don't add inflow to uba bundle state.
 
           // Throws errors if inflow is before, after, or in same bundle as outflow.
-          assertPromiseError(ubaClient.validateFlow(outflow), "Could not find matched deposit in same bundle");
+          void assertPromiseError(ubaClient.validateFlow(outflow), "Could not find matched deposit in same bundle");
```
Comment (mrice32) [inline test/UBAClient.validateFlow.ts:425]: why don't we await these promises?

File: src/utils/RedisUtils.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export async function disconnectRedisClient(logger?: winston.Logger): Promise<vo
   if (redisClient !== undefined) {
     // todo understand why redisClient isn't GCed automagically.
     logger.debug("Disconnecting from redis server.");
-    redisClient.disconnect();
+    await redisClient.disconnect();
```
Comment (mrice32) [inline src/utils/RedisUtils.ts:87]: Nice find!

**Review by mrice32 — APPROVED** [review]
Great addition!

File: test/UBAClient.validateFlow.ts (hunk 1 above)
Comment (james-a-morris) [reply test/UBAClient.validateFlow.ts:425]: These are in the tests and purely serve to throw errors

**Review by pxrl — APPROVED** [review]

**Pull request merged**